### PR TITLE
Replace `windows-sys` dependency with `windows-bindgen`

### PR DESCRIPTION
**Pull request by Jake-Shadle**
This PR replaces the `windows-sys` crate with only the minimal binding needed by this crate, removing a source of frustration https://github.com/microsoft/windows-rs/issues/1720

The currently published version uses the outdated 0.42 version, but as we can see in https://github.com/console-rs/console/pull/157, which bumps from 0.42 -> 0.45, the minor version churn of windows-sys is **irrelevant** for this crate since the bindings are **never** going to meaningfully change.

### CODE REVIEW

File: tests/windows_bindings.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Uncomment this to regenerate bindings
```
Comment (kennykerr) [inline tests/windows_bindings.rs:1]: I'd recommend running this test unconditionally. Then you can ensure that the bindings are always up to date and regressions haven't snuck in. You can use a diff check via CI to ensure the generated bindings haven't diverged from what's committed to your repo.

Comment (kennykerr) [reply tests/windows_bindings.rs:1]: It's definitely a lot more work to maintain this kind of dependency than simply using a Cargo dependency on the `windows-sys` crate. 😊

**Comment by mitsuhiko** [comment]
Sorry I did not see this. I'm not sure if I like the idea of maintaining those bindings.

**Comment by MarijnS95** [comment]
@mitsuhiko is this something you might want to reconsider now that all the tooling is officially published by Microsoft? There are no bindings to "maintain", they only need to be regenerated with the official tooling which is nowadays just a simple `api_gen` crate in the workspace.

**Pull request closed without merging**